inference/deepdive-model

```diff
@@ -0,0 +1,240 @@
+                esac
+                # keep a LAST symlink pointing to the last
+                mkdir -p snapshot/model/"$What"
+                ln -sfnv "$Name" snapshot/model/"$What"/LAST
+                ;;
+
+            reuse|drop)
+                if [[ $# -eq 0 ]]; then
+                    Name=LAST
+                    [[ -e "$DEEPDIVE_APP/snapshot/model/$What/$Name" ]] ||
+                        error "\`deepdive $What keep\` has never been run"
+                else
+                    Name=$1; shift
+                fi
+                # determine the actual artifact
+                if [[ -d "$Name" ]]; then
+                    src=$(cd "$Name" && pwd)
+                    cd "$DEEPDIVE_APP"
+                else
+                    cd "$DEEPDIVE_APP"
+                    src="snapshot/model/$What/$Name"
+                    [[ -e $src ]] ||
+                        error "\`deepdive keep $What $Name\` has never run or $Name got removed"
+                fi
+                case $Command in
+                    reuse)
+                        case $What in
+                            factorgraph)
+                                cd "$DEEPDIVE_APP"
+                                # TODO use (relative) symlink or cp -al instead of duplicating with rsync?
+                                rsync -avH --delete --copy-unsafe-links "$src"/factorgraph{,.done} run/model/
+                                # TODO load all table for grounding as well
+                                ;;
+                            weights)
+                                # TODO load into a unique table, then create a view with the following name
+                                weightTableName=dd_graph_weights_reuse
+                                weightsReuseFlag="$DEEPDIVE_APP"/run/model/grounding/factor/weights.reuse
+                                deepdive create table "$weightTableName" "description:TEXT" "categories:TEXT" "weight:DOUBLE PRECISION"
+                                rm -f "$weightsReuseFlag"
+                                for weightFile in "$src"/dd_weights{,.multinomial}.tsv.bz2; do
+                                    [[ -e "$weightFile" ]] || continue
+                                    case $weightFile in
+                                        *.multinomial.tsv.bz2)
+                                            # first load the grouped data from file
+                                            deepdive create table "${weightTableName}_array" "description:TEXT" "categories::TEXT[]" "weights:DOUBLE PRECISION[]"
+                                            # TODO parallel load
+                                            deepdive load "${weightTableName}_array" "$weightFile"
+                                            # explode the group
+                                            # XXX UNNEST is postgres-specific
+                                            deepdive sql "INSERT INTO $weightTableName
```

Comment:
Since we had more than one sources of weights (one for boolean, another for multinomial factors) I kept it as INSERT, but we could also create two tables and add a UNION view if you think there's a big difference.